osquery/events/windows/windowseventlogpublisher.cpp

```diff
@@ -0,0 +1,205 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/config/config.h>
+#include <osquery/events/windows/windowseventlogpublisher.h>
+#include <osquery/flags.h>
+#include <osquery/logger.h>
+#include <osquery/registry_factory.h>
+
+#include <plugins/config/parsers/feature_vectors.h>
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace {
+const int kCharFreqVectorLen{256};
+
+Status loadCharacterFrequencyMap(std::vector<double>& character_frequency_map) {
+  character_frequency_map = {};
+
+  auto parser = Config::getParser(kFeatureVectorsRootKey);
+  if (parser == nullptr) {
+    return Status::failure(
+        "Failed to acquire the feature_vectors configuration parser");
+  }
+
+  const auto& config = parser->getData().doc();
+  if (!config.IsObject()) {
+    return Status::failure("The configuration is not valid");
+  }
+
+  if (!config.HasMember("feature_vectors") ||
+      !config["feature_vectors"].HasMember("character_frequencies")) {
+    return Status::success();
+  }
+
+  if (!config["feature_vectors"]["character_frequencies"].IsArray()) {
+    return Status::failure(
+        "The character_frequencies configuration entity is not a valid array");
+  }
+
+  const auto& character_frequencies_array =
+      config["feature_vectors"]["character_frequencies"];
+  if (character_frequencies_array.Empty()) {
+    return Status::failure(
+        "The character_frequencies configuration entity array is empty");
+  }
+
+  std::vector<double> output(kCharFreqVectorLen, 0.0);
+  for (unsigned int i = 0; i < character_frequencies_array.Size(); i++) {
```

Comment:
I am not sure what `.Size()` is, it should be `size_t`, use that or `auto` instead of a `unsigned int` for portability.